File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
  
    vector<int> twoSum(vector<int>& nums, int target) {
     unordered_map<int,int>x;
        vector<int>v;
        for(int i=0;i<nums.size();i++)
        { 
            if(x.find(target-nums[i]) != x.end())
            {
                v.push_back(x[target-nums[i]]);
                v.push_back(i);
                return v;
            }
            x[nums[i]] = i;
        }
        return v;
    }
};

File: 0001-two-sum/0001-two-sum.java

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>map=new HashMap<>();
        int ans[]=new int[2];
        for(int i=0;i<nums.length;i++)
        {
            int y=target-nums[i];
            if(map.containsKey(y))
            {
                ans[0]=map.get(y);
                ans[1]=i;
                return ans;
            }
            else
                map.put(nums[i],i);
            
        }
        return ans;
    }
}